### DISCUSSION TITLE
deepin应用加入csd模式支持。

### DISCUSSION THREAD

**Discussion by chenchongbiao** [linuxdeepin/developer-center#4637] [Features Request & Ideas | 特性请求 & 头脑风暴]
WSLg可以在Windows系统下直接使用WSL子系统上安装图形应用。

问题：
WSLg下使用的应用是有Weston窗管进行管理的，那么在WSL上安装deepin 23系统，在wayland模式下，使用如deepn-terminal应用，会给应用绘制窗口的边框和标题栏，命令`deepin-terminal -platform dwayland`。
效果：
![图片](https://github.com/linuxdeepin/developer-center/assets/52627687/668a73f4-8f28-4569-ad94-c3c625804673)

在dtkwidget中，dmainwindow的构造函数中加入setWindowFlag(Qt::FramelessWindowHint);取消边框和标题栏的绘制。
期望效果：
![图片](https://github.com/linuxdeepin/developer-center/assets/52627687/31f2d68f-90cb-4893-b96b-3393805866e4)

需要考虑代码的兼容性问题，发起讨论如何进行修改让WSLg的wayland模式下，可以不显示边框和标题栏。

**Comment by chenchongbiao** [comment]
@zccrs

**Comment by zccrs** [comment]
@kegechen 可以考虑在 platform plugin 中实现
